starfish/src/main/java/org/maxandersen/Starfish.java

```diff
@@ -0,0 +1,238 @@
+package org.maxandersen;
+
+import io.quarkus.runtime.QuarkusApplication;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.io.File;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+public class Starfish implements QuarkusApplication {
+  @Override
+  public int run(String... args) throws Exception {   
+    
+    
+    for(String s:args)
+    System.out.println("\nTarget : "+s);//Printing Received Arguements for Debugging
+
+    if(args[0].equalsIgnoreCase("config")){
+    //Incase user wants to configure starfish
+    editConfig();//Calling function that enables to edit configuration
+    return 10; //Incase user typed "starfish config" we only want to edit configuration
+    }
+
+    //Calling function for Fetching installed Packages 
+    //fetch_installed_packages(); //This function will print all installed packages
+
+    //Configuration identifiers
+    String clone_path="";//Holds path to  destination Where the Repository Must Be Clonned
+    String ide="";  //Holds command for IDE to open upon
+    String userHome = System.getProperty( "user.home" ); //Get User Home Directory: /home/user_name
+
+    File configFile = new File(userHome + "/starfish-config.yml"); //Loading YAML
+
+
+  
+    
+    
+    //Reading Configuration File
+    try {
+        if (!configFile.exists()) {
+            //Incase no config file exists
+            System.out.println("\nThis is the first time you're using starfish...You'll need to  configure it");
+            editConfig(); //Calling function that lets user to configure 
+        }
+        
+        System.out.println("\nLoading configurations.....\n");
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    
+        Config config = mapper.readValue(configFile, Config.class);
+        
+        ide=config.ide;
+        clone_path=config.clone_path;
+        if(ide==null||clone_path==null)editConfig(); //Incase of absence of configuration in file Launch Config 
+        
+        
+    }
+    catch (Exception e) {
+        e.printStackTrace();
+        System.out.print("Press any key to continue . . . ");
+    }
+    
+
+
+    //Clonnning Git Repo
+    //Expected parameter: https://github.com/user-name/repo-name.git
+    String repo_name=args[0].substring(args[0].lastIndexOf("/"),args[0].lastIndexOf(".")); //Extracts the Name of Repository
+    //String clone_path= "/home/fahad/MyProjects/starfish_clonned/"; 
+    String cmd = "git clone "+args[0];
```

Comment:
This is taking a random unvalidated string and then executed it raw on users machine. This is too risky to expose. 

Pretty sure you can add line "; rm -rf *" and bad things would happen. 

Better done by ensuring you actually have a valid URL and execute with separate arguments rather than one long string.

Reply:
`String cmd = "git clone "+args[0];` : was residual code.I wasn't using this anyhow.
Separate functions are being used:
- `public static void gitClone(Path directory, String originUrl)` 
- `public static void runCommand(Path directory, String... command)`

**URL validation is very important.**
I'll implement it soon.
Thanks :)

Reply:
Done @https://github.com/maxandersen/starfish/pull/8/commits/83ef8e5add5aa885402e5d3f701b4bb6c01c5255